### ISSUE TITLE
ResultSet.toJson() does not output correct ISO 8601 dates or times

### ISSUE THREAD

**Issue by hyyan** [BasisHub/components#164]
According to[ ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format, you must specify each element in the format. This includes the T, the colons (:), the + or - , and the periods (.). currently, the `ResultSetJsonMapper` drops the time zone designator for dates and drops the date element plus the time zone designator for times. 

| Wrong               | Correct                                                                   | Notes                                                                                                                                                                                                                                                                                             |
|---------------------|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1979-08-03T00:00:00 | 1979-08-03T00:00:00Z 1979-08-03T00:00:00+09:00 1979-08-03T00:00:00-01:00  | ISO 8601 specifies that time without a  Z, and without an offset (such as  -08:00) should be treated as  local time. which means the browser will auto adjust the time in this case to fit the user timezone                                                                                      |
| 05:30:06            | 1979-08-03T05:30:06Z 1979-08-03T05:30:06+09:00  1979-08-03T05:30:06-01:00 | ISO 8601 time is one element of the specification, and it needs a date attached to be parsable. In the grid, once we receive a time string with a timezone designator we complete the missing date part with a static predefined date (1970-01-0). so  `05:30:06Z` becomes `1970-01-01T05:30:06Z` |

**Comment by Frank-en-stein** [comment]
Could you specify which `java.sql.Types` you used? I did something like this to test and it happens to work fine:
```
DataRow dr = new DataRow();
dr.setFieldValue("Date", java.sql.Types.TIME_WITH_TIMEZONE, "1979-08-03T00:00:00+09:00");
```

**Comment by hyyan** [comment]
@Frank-en-stein the types don't matter. Take a look at [this code](https://github.com/BasisHub/components/blob/master/src/main/java/com/basiscomponents/db/util/ResultSetJsonMapper.java#L284-L304)

When dates are parsed, the `Z` is dropped and when times are parsed the date part is dropped. passing these incomplete ISO 8601 date-times  to the browser will make the browser thinks that these
are local date-times and not UTC zoned date-times

Even [this part of the code](https://github.com/BasisHub/components/blob/master/src/main/java/com/basiscomponents/db/util/ResultSetJsonMapper.java#L244) is wrong which assumes the date-times must be zoned according to the default JRE timezone. @StephanWald This is something that needs to be configured in a global config object along with other stuff. The ExGrid should also be able to access this information so we can customize renderers in a way that honors a specific language, timezone. mask ...

**Comment by Frank-en-stein** [comment]
So, just `java.sql.Types.DATE` should reflect the ISO 8601 behavior?

**Comment by hyyan** [comment]
@Frank-en-stein 
`java.sql.Types.DATE`
`java.sql.Types.TIME`
`java.sql.Types.TIME_WITH_TIMEZONE`
`java.sql.Types.TIMESTAMP`
`java.sql.Types.TIMESTAMP_WITH_TIMEZONE`

**Comment by Frank-en-stein** [comment]
Please correct me if I'm wrong. The fix should be parsing the provided string representing date with all the fields required by ISO 8601 even if some field (say timezone) isn't mentioned. The omitted fields should be parsed with discussed default value, right?

Now if the aforementioned is correct, doesn't it make the output of all the `java.sql.Types` same?

**Comment by StephanWald** [comment]
@Frank-en-stein, maybe let's do an intermediate step: please create a seeder package (find a good place for it) that has place for a ResultSetSeeder and a DataRowSeeder, and then create a method therein that returns a DataRow which has a field for each of the java.sql.Types types that could theoretically happen, with a fixed sample value. Then we create a unit test for the toJson, where we can discuss and define the expected JSON Output for all SQL types, once and forever.

See https://github.com/BasisHub/components/issues/72 for the seeder idea.

**Comment by Frank-en-stein** [comment]
@StephanWald got it. I'll check back with you as soon as I make some time to figure out the implementation details.

**Comment by hyyan** [comment]
@StephanWald , @Frank-en-stein Any update on this one?

**Comment by Frank-en-stein** [comment]
Hi,
Actually. I'm a bit preoccupied. It might take me some time to get back to
the task as I'm gonna need to setup everything on a new machine.

Kind regards

**Comment by StephanWald** [comment]
[{
        "SOMETIMESTAMP": "2020-03-10T16:40:17.332+01:00",
        "SOMEDATE": "2020-03-10T00:00:00+01:00",
        "SOMETIME": "1970-01-01T16:40:17+00:00",
        "meta": {
            "SOMETIMESTAMP": {
                "ColumnType": "93"
            },
            "SOMEDATE": {
                "ColumnType": "91"
            },
            "SOMETIME": {
                "ColumnType": "92"
            }
        }
    }
]

**Comment by hyyan** [comment]
Fixed in BBj 20.10
